web/ui/ui.go

```diff
@@ -1,12 +1,119 @@
 package ui
 
 import (
+	"bytes"
+	"encoding/base64"
+	"github.com/labbsr0x/goh/gohtypes"
+	"html/template"
+	"io/ioutil"
+	"mime/quotedprintable"
 	"net/http"
+	"os"
 	"path"
 )
 
+// Enum
+const (
+	Consent           = "consent.html"
+	EmailConfirmation = "email_confirmation.html"
+	Layout            = "index.html"
+	Login             = "login.html"
+	Registration      = "registration.html"
+	Update            = "update.html"
+)
+
+type IPage interface {
+	SetHTML(html template.HTML)
+}
+
 // Handler defines the handler for ui requests
 func Handler(baseUIPath string) http.Handler {
 	uiPath := path.Join(baseUIPath, "")
 	return http.FileServer(http.Dir(uiPath))
 }
+
+// BuildPage is used to load a page with the standard layout
+func BuildPage(baseUIPath, htmlFile string, page IPage) []byte {
+	buf := new(bytes.Buffer)
+	content := template.Must(template.ParseFiles(path.Join(baseUIPath, htmlFile)))
+
+	err := content.Execute(buf, page)
+	gohtypes.PanicIfError("Unable to load page", http.StatusInternalServerError, err)
+
+	html, err := ioutil.ReadAll(buf)
+	gohtypes.PanicIfError("Unable to read page", http.StatusInternalServerError, err)
+
+	page.SetHTML(template.HTML(html))
+
+	layout := template.Must(template.ParseFiles(path.Join(baseUIPath, Layout)))
+	err = layout.Execute(buf, page)
+	gohtypes.PanicIfError("Unable to load layout", http.StatusInternalServerError, err)
+
+	return buf.Bytes()
+}
+
+func BuildMail(baseUIPath, htmlFile string, mailContent interface{}) []byte {
+	var body bytes.Buffer
+
+	boundary := "f46d043c813270fc6b04c2d223da"
```

Comment:
@claudiosegala, shouldn't this hash change with every email being built? Also, this seems specific for email. Why isn't it in the `mail` package?

Reply:
Since the HTML from emails is together with the HTML of pages. I thought that the building of the mail should come together with the build of the pages

Reply:
Well, if you take a look at the `BuildPage` code it is very generic. Any page could be built with it. The `BuildMail` on the other hand is very specific for the email confirmation issue. So I reiterate, I think it makes more sense to send this to mail.

Reply:
BuildMail is also generic, it receives a filename from where to get the html and a page to fill it. But I will transfer it to the mail package.